README.md

```diff
@@ -1,70 +1,220 @@
+Example:
 
-One of the Well Known parties, is the **Public** party that can be used to disclose information via contracts to all participants.
-```typescript
-// Within a
-<PublicLedger ledgerId={ledgerId}>
-  ...
-  // The standard hooks provided by @daml/react have an analog "AsPublic" method.
-  // Eg. instead of useStreamQuery one can write:
-  var contracts = useStreamQueryAsPublic(Contract);
-
-</PublicLedger>
+```tsx
+<DamlHubLogin onLogin={(credential, error) => {
+  if (credential) {
+    /* handle credentials. the application may cache the token and use it for subsequent API calls. */
+    const { party, token } = credential;
+    console.log(party, token);
+  }
+  else { /* handle error */  }
+}}>
 ```
 
-### Parties Login
+The `credential` object returned in the callback is an instance of the `PartyToken` class. The application may check the validity of a token at a later point by using the getter `credential.isExpired`.
 
-For standard applications, we recommend using the login with button endpoint, a simple anchor tag that links to the DABL login endpoint (as described [here](https://docs.projectdabl.com/api/iam/#the-login-in-with-dabl-button)). The endpoint will redirect back to your application with a JWT access token and party ID, and is the right way for **real production users** to log in to your application.
+#### Login Methods
 
-For development purposes, it may be useful to be able to supply a list of party credentials for easy login, to test your app from the point-of-view of different parties. You can do so with the `DablPartiesInput` component, which allows you to load in the `parties.json` file. This file can be downloaded from the Users tab of the ledger in the project:DABL management console.
+For development purposes, it may be wise to test the application with multiple different parties with different roles. In this case, it is possible to specify alternate login methods. There are three methods altogether, with short-form props that can be used to enable them in a mix-and-match fashion:
 
-The file is not uploaded anywhere, but is parsed in-browser via the [FileReader APIs](https://developer.mozilla.org/en-US/docs/Web/API/FileReader). Note that access tokens expire every 24 hours.
+1. `withButton`: Button login. This is the default if no login method props are given. It is the interactive user login flow as described above, for production applications. Note that if either of the two other method props are supplied, the button is then disabled unless `withButton` is explicitly given alongside them.
 
-```typescript
-<DablPartiesInput
-  ledgerId='ledger-xyz'
-  onLoad={
-    parties => parties.map(p => console.log(`Party ID: ${p.party}, party token: ${p.token}`))
-  }
+2. `withFile`: Login with a `parties.json` file. This is a file containing a list of all the parties (and associated ledger access tokens) that you've created under your Daml Hub account for the ledger. It can be downloaded from Daml Hub.
```

Comment:
I noticed you start to use the second person here whereas previously a more passive voice. I think using the second person is more clear - so take it or leave it but just an observation! :)